lint-rules/src/main/java/com/duckduckgo/lint/ui/DaxSecureTextFieldTrailingIconDetector.kt

```diff
@@ -0,0 +1,113 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.lint.ui
+
+import com.android.tools.lint.client.api.UElementHandler
+import com.android.tools.lint.detector.api.Category.Companion.CUSTOM_LINT_CHECKS
+import com.android.tools.lint.detector.api.Detector
+import com.android.tools.lint.detector.api.Implementation
+import com.android.tools.lint.detector.api.Issue
+import com.android.tools.lint.detector.api.JavaContext
+import com.android.tools.lint.detector.api.Scope
+import com.android.tools.lint.detector.api.Severity
+import com.android.tools.lint.detector.api.SourceCodeScanner
+import com.android.tools.lint.detector.api.TextFormat
+import org.jetbrains.uast.UCallExpression
+import org.jetbrains.uast.getParameterForArgument
+import java.util.EnumSet
+
+@Suppress("UnstableApiUsage")
+class DaxSecureTextFieldTrailingIconDetector : Detector(), SourceCodeScanner {
+
+    override fun getApplicableUastTypes() = listOf(UCallExpression::class.java)
+
+    override fun createUastHandler(context: JavaContext): UElementHandler = DaxSecureTextFieldCallHandler(context)
+
+    internal class DaxSecureTextFieldCallHandler(private val context: JavaContext) : UElementHandler() {
+        override fun visitCallExpression(node: UCallExpression) {
+            val methodName = node.methodName
+
+            if (methodName == "DaxSecureTextField") {
+                checkTrailingIconParameter(node)
+            }
+        }
+
+        private fun checkTrailingIconParameter(node: UCallExpression) {
+            // Find the 'trailingIcon' parameter
+            val trailingIconArgument = node.valueArguments.find { arg ->
+                val parameterName = node.getParameterForArgument(arg)?.name
+                parameterName == "trailingIcon"
+            } ?: return // No 'trailingIcon' parameter provided which is fine
+
+            // Check if the trailingIcon uses an invalid composable
+            if (isInvalidComposable(trailingIconArgument)) {
+                reportInvalidComposableUsage(trailingIconArgument)
+            }
+        }
+
+        private fun isInvalidComposable(argument: org.jetbrains.uast.UExpression): Boolean {
+            val source = argument.sourcePsi?.text ?: return false
+
+            // Only DaxTextFieldTrailingIcon should be used in the trailingIcon parameter
+            // If the source doesn't contain it, then it's invalid
+            return !source.contains("DaxTextFieldTrailingIcon")
```

Comment:
We can probably get member list declared in the scope object class and check if the source contains one member of this list? This will allow us to avoid to edit this linter rule every time we need to edit the scope.